File: 设计模式/程序/10-DecoratorPattern/DecoratorPattern(C#)/Shape.cs
using System;


namespace DecoratorPattern
{
    public interface Shape
    {
        void draw();
    }

    public class Circle : Shape
    {
        public void draw()
        {
            Console.WriteLine("Draw a Circle");
        }
    }

    public class Rectangle : Shape
    {
        public void draw()
        {
            Console.WriteLine("Draw a Rectangle");
        }
    }
}

File: 设计模式/程序/10-DecoratorPattern/DecoratorPattern(python)/Shape.py
import abc

# 形状基类
class Shape(metaclass = abc.ABCMeta):
    @abc.abstractmethod
    def draw(self):
        pass

# 具体实现类
class Circle(Shape):
    def __init__(self, radius):
        self.radius = radius

    def draw(self):
        print("Draw a Circle with", self.radius, "radius")


class Rectangle(Shape):
    def __init__(self, length, width):
        self.length = length
        self.width = width

    def draw(self):
        print("Draw a Rectangle with", self.length, "length and", self.width, "width")
 
# 继续增加形状...
